keanu-project/src/main/java/io/improbable/keanu/vertices/VertexLabel.java

```diff
@@ -1,30 +1,58 @@
 package io.improbable.keanu.vertices;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+
 public class VertexLabel {
-    private final String label;
+    private static final char NAMESPACE_SEPARATOR = '.';
+    private static final char NAME_PREPENDER = ':';
```

Comment:
Is there a reason you've used a different value for the two separators?  If we went with something more "file system like" then just using the same '/' char would work for both perhaps?

Reply:
No major reason other than to distinguish more clearly between the namespace and the name.

I have no strong opinion, and these are only used in `toString`.

Reply:
Our toString() could well be useful for other stuff though - not least of which debug could be an important consumer